server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala

```diff
@@ -300,7 +300,7 @@ trait EmailQueryMethodContract {
         AppendCommand.builder().build(message2))
       .getMessageId
 
-    val requestDateMessage3 = Date.from(ZonedDateTime.now().minusDays(1).toInstant)
+    val requestDateMessage3 = Date.from(ZonedDateTime.now().minusDays(3).toInstant)
```

Comment:
Ok, so now we have message3 at t - 3d, message2 at t -2d ...
And the message are sorted like message3, message2, ...
So by ascending order of sent at. Not descending order as the name of the test and the request indicate.

You should change the assertion. Once changed, I think that the distributed version should work.
To make the memory version work you would need to add the sort by sent at in the SimpleMessageSearchIndex

Reply:
add a case for 'SentAt' in SimpleMessageSearchIndex.SearchComparator to make the sort work with the memory version

Reply:
> add a case for 'SentAt' in SimpleMessageSearchIndex.SearchComparator to make the sort works with the memory version

That's done: sortOnSentDateShouldWork had been re-enabled